lib/src/tasks/coverage/cli.dart

```diff
@@ -80,18 +89,27 @@ class CoverageCli extends TaskCli {
             'This project does not specify any integration tests.');
       }
     }
+    if(functional_tests.isEmpty){
+      if(functional && config.test.functionalTests.isEmpty){
+        return new CliResult.fail(
+            'This project does not specify any functional tests.');
+      }
+    }
```

Comment:
This if block should be nested within the above `if (tests.isEmpty) {` block because we only want to fail when no tests were specified, whereas this will fail when no functional tests are specified, even if other tests (unit, integration) were.